docs/route-tasks.md

```diff
@@ -0,0 +1,171 @@
+
+## Task routing with Task Queues
+
+Task Queues are what enable "Task routing". Task routing is the development technique that pairs Tasks with Workers. The basics of it can be broken down into 3 steps:
+
+1. Create a Worker that is registered to handle Workflow and/or Activity Tasks.
+2. Subscribe the Worker to listen to a Task Queue.
+3. Configure Workflows and Activities to send Tasks to the Task Queue.
+
+Let's look at the ["Hello World" sample in Go](https://github.com/temporalio/temporal-go-samples/tree/master/helloworld) to see how this is done.
+
+In the following code snippet, we [create a new Worker](https://github.com/temporalio/go-samples/blob/master/helloworld/worker/main.go#L26) that is subscribed to the "hello-world" Task Queue and is registered to handle the `helloworld.Workflow` and `helloworld.Activity`:
+
+```go
+// Worker is created and subscribed to the "hello-world" Task Queue
+w := worker.New(c, "hello-world", worker.Options{})
+defer w.Stop()
+// Worker is registered to handle the helloworld Workflow
+w.RegisterWorkflow(helloworld.Workflow)
+// Worker is registered to handle the helloworld Activity
+w.RegisterActivity(helloworld.Activity)
+```
+
+In the next code snippet, we [start the `helloworld.Workflow`](https://github.com/temporalio/go-samples/blob/master/helloworld/starter/main.go#L25):
+
+```go
+// Workflow Options specify the "hello-world" Task Queue
+// Which is the Task Queue that the Workflow will send its Tasks to
+workflowOptions := client.StartWorkflowOptions{
+  ID:        "hello_world_workflowID",
+  TaskQueue: "hello-world",
+}
+// Workflow Options are passed to the execution of the Workflow
+we, err := c.ExecuteWorkflow(
+  context.Background(), 
+  workflowOptions, 
+  helloworld.Workflow, 
+  "Temporal",
+)
+```
+
+:::caution
+
+If a Worker pulls a Task that it has not been registered to run, an "unknown workflow type" error will occur.
+
+:::
+
+## Task routing for Activity dependencies
+
+The Temporal service will persist Activity results as long as the results are less than 2MB in size and can be serialized for storage (which happens to be true for most Activity results). However, there are some scenarios where an Activity execution will do something that results in data, such as a large file, being stored directly on the host where the Worker is running. It is also possible that the next Activity in the Workflow has a dependency on that data and needs to run on the same host.
+
+Let's look at a file processing Workflow example where files are downloaded, processed in some way, and then uploaded to some other location. If a large video file is being processed only the name of the file can persist normally. The file itself must persist locally on the Worker host, as it is too big for normal Temporal storage. This means that any future Activities that act on the video file must also take place on the host where the Worker downloaded it.
```

Comment:
I think it might make sense to frontload this before the actual code etc. Right now this doc is very abstract until this section. I understand its a bit tricky because the problem itself isn't clear until you understand the limit on activities etc.

Reply:
Also "normal Temporal storage" is too generic

Reply:
"processed" with "transcoded"